File: tests/fixtures/calls_sample.py
"""Fixture for testing CALLS edge extraction."""


def helper() -> str:
    """A simple helper function."""
    return "ok"


def main() -> None:
    """Calls helper() — should produce a CALLS edge main→helper."""
    result = helper()
    return result


class Service:
    """A service with methods that call each other."""

    def process(self, data: str) -> str:
        """Calls self.validate() — should produce CALLS process→validate."""
        if self.validate(data):
            return self.transform(data)
        return ""

    def validate(self, data: str) -> bool:
        """Validates data."""
        return bool(data)

    def transform(self, data: str) -> str:
        """Transforms data."""
        return data.upper()


class Pipeline:
    """Uses Service — should produce CALLS edges to Service constructor."""

    def run(self, value: str) -> str:
        """Calls Service() constructor and process()."""
        svc = Service()
        return svc.process(value)

File: tests/fixtures/calls_sample.ts
// Fixture for testing CALLS edge extraction in TypeScript

function helper(): string {
    return "ok";
}

function main(): void {
    // Calls helper() — should produce a CALLS edge main→helper
    const result = helper();
}

class Service {
    process(data: string): string {
        // Calls this.validate() and this.transform()
        if (this.validate(data)) {
            return this.transform(data);
        }
        return "";
    }

    validate(data: string): boolean {
        return data.length > 0;
    }

    transform(data: string): string {
        return data.toUpperCase();
    }
}

// Arrow function that calls helper() — tests arrow CALLS extraction
export const pipeline = (input: string): string => {
    return helper();
};
